example-card-integration/data-connector/actions/patchaction.ts

```diff
@@ -0,0 +1,75 @@
+import {PatchChange, SerializedIsoDateObject} from 'lucid-extension-sdk';
+import {DataConnectorPatchAction} from 'lucid-extension-sdk/dataconnector/actions/action';
+import {TaskFieldNames} from '../../common/names';
+import {ExternalExampleDataClient} from '../../data/exampleexternaldataclient';
+import {ExampleTaskType} from '../schema/examplecollectionschema';
+import {fullTaskDataToLucidFormat} from '../schema/exampletaskutil';
+
+export const patchAction: (action: DataConnectorPatchAction) => Promise<PatchChange[]> = async (action) => {
+    const extenalExampleDataClient = new ExternalExampleDataClient();
+    return await Promise.all(
+        action.patches.map(async (patch) => {
+            const change = patch.getChange();
+            await Promise.all([
+                ...Object.entries(patch.itemsAdded).map(async ([oldPrimaryKey, additions]) => {
```

Comment:
Nit: this huge promise.all statements gets pretty nested. Consider breaking down into helpers